services/alertmanager/registry.go

```diff
@@ -93,7 +115,9 @@ func (r *Registry) collect() ammodels.PostableAlerts {
 	now := now()
 	for id, t := range r.times {
 		if t.Before(now) {
-			res = append(res, r.alerts[id])
+			alert := r.alerts[id]
+			alert.EndsAt = strfmt.DateTime(now.Add(resolveTimeoutFactor * r.resendInterval))
```

Comment:
`resolveTimeoutFactor * r.resendInterval` better precalculate that value in the constructor as it never changed after service start. You can name it `alertTTL` for example.

Reply:
done